chaste_codegen/_lookup_tables.py

```diff
@@ -0,0 +1,207 @@
+import collections
+
+from cellmlmanip.model import Variable
+from sympy import (
+    acos,
+    acosh,
+    acot,
+    acoth,
+    acsc,
+    acsch,
+    asec,
+    asech,
+    asin,
+    asinh,
+    atan,
+    atanh,
+    cos,
+    cosh,
+    cot,
+    coth,
+    csc,
+    csch,
+    exp,
+    ln,
+    log,
+    sec,
+    sech,
+    sin,
+    sinh,
+    tan,
+    tanh,
+)
+
+from chaste_codegen._math_functions import (
+    acos_,
+    cos_,
+    exp_,
+    sin_,
+)
+from chaste_codegen._rdf import OXMETA
+
+
+# The expensive functions the lookup table analysis searches for
+_EXPENSIVE_FUNCTIONS = (exp, log, ln, sin, cos, tan, sec, csc, cot, sinh, cosh, tanh, sech, csch, coth, asin, acos,
+                        atan, asinh, acosh, atanh, asec, acsc, acot, asech, acsch, acoth, exp_, acos_, cos_, sin_)
+
+
+# tuple of ([<metadata tag>, mTableMins, mTableMaxs, mTableSteps], )
+_DEFAULT_LOOKUP_PARAMETERS = (['membrane_voltage', -250.0001, 549.9999, 0.001], )
+
+
+class LookupTables:
+    """ Holds information about lookuptables and methods to analyse the model for lookup tables.
+    """
+
+    def __init__(self, model, lookup_params=_DEFAULT_LOOKUP_PARAMETERS):
+        """ Initialise a LookUpTables instance
+        :param model: A :class:`cellmlmanip.Model` object.
+        :param lookup_params: Optional collection of lists: [[<metadata tag>, mTableMins, mTableMaxs, mTableSteps]]
+        """
+        # Lookup vars is a tuple of [<metadata tag>, mTableMins, mTableMaxs, mTableSteps,
+        #                            <set_of_method_names_table_is_used_in>, variable, [<lookup epxrs>]]
+        self._lookup_parameters = list(map(lambda p: p + [set(), None, []], lookup_params))
```

Comment:
I'd be inclined to make this a list of dicts rather than remembering which indices mean what

Reply:
that's a really good idea actually, why didn't I think of that :)